### ISSUE TITLE
java.lang.IncompatibleClassChangeError: Found class org.jetbrains.kotlin.ir.declarations.IrModuleFragment, but interface was expected

### ISSUE THREAD

**Issue by andylamax** [bnorm/kotlin-react-function#15]
I am getting this error just when I include the gradle plugin
```kotlin
plugins {
    kotlin("js") version "1.4.21" // also tried version "1.4.10"
    id("com.bnorm.react.kotlin-react-function") version "0.1.0" // also tried version "0.2.1"
}
```

When gradle is executing task `compileKotlinJs` I get the error
```
e: java.lang.IncompatibleClassChangeError: Found class org.jetbrains.kotlin.ir.declarations.IrModuleFragment, but interface was expected
        at com.bnorm.react.ReactFunctionIrGenerationExtension.generate(ReactFunctionIrGenerationExtension.kt:25)
        at org.jetbrains.kotlin.ir.backend.js.KlibKt$generateModuleFragmentWithPlugins$1.invoke(klib.kt:331)
        at org.jetbrains.kotlin.ir.backend.js.KlibKt$generateModuleFragmentWithPlugins$1.invoke(klib.kt)
        at org.jetbrains.kotlin.psi2ir.Psi2IrTranslator.generateModuleFragment(Psi2IrTranslator.kt:95)
        at org.jetbrains.kotlin.ir.backend.js.KlibKt.generateModuleFragmentWithPlugins(klib.kt:336)
        at org.jetbrains.kotlin.ir.backend.js.KlibKt.generateKLib(klib.kt:170)
        at org.jetbrains.kotlin.cli.js.K2JsIrCompiler.doExecute(K2JsIrCompiler.kt:190)
        at org.jetbrains.kotlin.cli.js.K2JSCompiler.doExecute(K2JSCompiler.java:181)
        at org.jetbrains.kotlin.cli.js.K2JSCompiler.doExecute(K2JSCompiler.java:74)
        at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:88)
        at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:44)
        at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:98)
        at org.jetbrains.kotlin.incremental.IncrementalJsCompilerRunner.runCompiler(IncrementalJsCompilerRunner.kt:174)
        at org.jetbrains.kotlin.incremental.IncrementalJsCompilerRunner.runCompiler(IncrementalJsCompilerRunner.kt:70)
        at org.jetbrains.kotlin.incremental.IncrementalCompilerRunner.compileIncrementally(IncrementalCompilerRunner.kt:249)
        at org.jetbrains.kotlin.incremental.IncrementalCompilerRunner.access$compileIncrementally(IncrementalCompilerRunner.kt:38)
        at org.jetbrains.kotlin.incremental.IncrementalCompilerRunner$compile$2.invoke(IncrementalCompilerRunner.kt:80)
        at org.jetbrains.kotlin.incremental.IncrementalCompilerRunner.compile(IncrementalCompilerRunner.kt:92)
        at org.jetbrains.kotlin.daemon.CompileServiceImplBase.execJsIncrementalCompiler(CompileServiceImpl.kt:547)
        at org.jetbrains.kotlin.daemon.CompileServiceImplBase.access$execJsIncrementalCompiler(CompileServiceImpl.kt:93)
        at org.jetbrains.kotlin.daemon.CompileServiceImpl.compile(CompileServiceImpl.kt:1732)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:357)
        at sun.rmi.transport.Transport$1.run(Transport.java:200)
        at sun.rmi.transport.Transport$1.run(Transport.java:197)
        at java.security.AccessController.doPrivileged(Native Method)
        at sun.rmi.transport.Transport.serviceCall(Transport.java:196)
        at sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:573)
        at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:834)
        at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(TCPTransport.java:688)
        at java.security.AccessController.doPrivileged(Native Method)
        at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:687)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

**Comment by andylamax** [comment]
Trying out version "0.2.0" the one available in the sample. Works

**Comment by andylamax** [comment]
Nope scratch. It doesn't work form me

**Comment by bnorm** [comment]
Kotlin 1.4.20+ is not supported by this compiler plugin, yet. The incompatible change error looks to be because Kotlin changed `IrModuleFragment` from an interface to a class. [Support for 1.4.20+ is in the works](https://github.com/bnorm/kotlin-react-function/pull/14) (slowly) but in the meantime you will need to stick with Kotlin 1.4.10 to use this plugin.

**Comment by bnorm** [comment]
Fixed in release `0.3.0`.

**Comment by andylamax** [comment]
Much thanks
